internal/pkg/cachecli/cache_list.go

```diff
@@ -0,0 +1,197 @@
+			if err != nil {
+				sylog.Warningf("Unable to get stat: %v", err)
+				return err
+			}
+			fmt.Printf("%-22s %-22s %-16s %s\n", c.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), findSize(fileInfo.Size()), "library")
+		}
+	}
+	return nil
+}
+
+func listOciCache() error {
+	// loop thrught oci-tmp cache
+	ociTmp, err := ioutil.ReadDir(cache.OciTemp())
+	if err != nil {
+		sylog.Warningf("Unable to opening oci-tmp folder: %v", err)
+		return err
+	}
+	for _, f := range ociTmp {
+		blob, err := ioutil.ReadDir(filepath.Join(cache.OciTemp(), f.Name()))
+		if err != nil {
+			sylog.Warningf("Unable to looking in cache: %v", err)
+			return err
+		}
+		for _, b := range blob {
+			fileInfo, err := os.Stat(filepath.Join(cache.OciTemp(), f.Name(), b.Name()))
+			if err != nil {
+				sylog.Warningf("Unable to get stat: %v", err)
+				return err
+			}
+			fmt.Printf("%-22s %-22s %-16s %s\n", b.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), findSize(fileInfo.Size()), "oci")
+		}
+	}
+	return nil
+}
+
+func listBlobCache(printList bool) error {
+	// loop thrught ociBlob cache
+	count := 0
+	var totalSize int64
+
+	_, err = os.Stat(filepath.Join(cache.OciBlob(), "/blobs"))
+	if os.IsNotExist(err) {
+		return nil
+	}
+	blobs, err := ioutil.ReadDir(filepath.Join(cache.OciBlob(), "/blobs/"))
+	if err != nil {
+		sylog.Warningf("Unable to opening oci folder: %v", err)
+		return err
+	}
+	for _, f := range blobs {
+		blob, err := ioutil.ReadDir(filepath.Join(cache.OciBlob(), "/blobs/", f.Name()))
+		if err != nil {
+			sylog.Warningf("Unable to looking in cache: %v", err)
+			return err
+		}
+		for _, b := range blob {
+			fileInfo, err := os.Stat(filepath.Join(cache.OciBlob(), "/blobs/", f.Name(), b.Name()))
+			if err != nil {
+				sylog.Warningf("Unable to get stat: %v", err)
+				return err
+			}
+			if printList == true {
+				fmt.Printf("%-22.20s %-22s %-16s %s\n", b.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), findSize(fileInfo.Size()), "blob")
+			}
+			count++
+			totalSize += fileInfo.Size()
+		}
+	}
+	if printList != true && count >= 1 {
+		fmt.Printf("\nThere are %d oci blob file(s) using %v of space. Use: '-T=blob' to list\n", count, findSize(totalSize))
+	}
+	return nil
+}
+
+// ListSingularityCache : list local singularity cache, typeNameList : is a string of what cache
+// to list (seprate each type with a comma; like this: library,oci,blob) allList : force list all cache.
+func ListSingularityCache(typeNameList string, allList bool) error {
+	libraryList := false
+	ociList := false
+	blobList := false
+	listBlobSum := false
+
+	// split the string for each `,` then loop throught it and find what flags are there.
+	// then see whats true/false later. heres the benefit of doing it like this; if the user
+	// specified `library` twice, it will still only be printed once.
+	if len(typeNameList) >= 1 {
+		for _, nameType := range strings.Split(typeNameList, ",") {
+			if nameType == "library" {
```

Comment:
I would replace these if-else blocks with a switch-case statement:

```
switch nameType {
    case "library":
    case "oci":
    [...]
    // `default:` signals what to do if none of the above were found, here it would be filling the function of your final else block.
    default: 
}
```